src/views/collection-view/server.ts

```diff
@@ -27,14 +27,29 @@ export interface CollectionPageProps {
 export type CollectionLayout = Pick<
   SidebarSidecarLayoutProps,
   'headings' | 'breadcrumbLinks'
-> & { sidebarSections: CollectionCategorySidebarSection[] }
+> & { collectionViewSidebarSections: CollectionCategorySidebarSection[] }
 
 export interface CollectionPagePath {
   params: {
     collectionSlug: string
   }
 }
 
+export async function getCollectionViewSidebarSections(
+  product: LearnProductData,
```

Comment:
💭  This could probably just be `productSlug`, to limit the data being passed there?

Reply:
I think this would be the function that we would call client-side on menu open if we wanted to do that dynamic approach.... then the sidebar children could be generated based on that data.

Reply:
Riffing off of my other comment proposing something like `TutorialsSidebarContent`, I wonder if we could call this function from there? Doing that should defer the call until the component is rendered. For this view, that'll be immediately since the level that needs the data is at the level of this view. For `TutorialView`, I think `TutorialsSidebarContent` wouldn't be rendered until the `Sidebar` is opened and the level up button is pressed. 🤔 I'll experiment with this a little bit when I make `TutorialsSidebarContent`!

Reply:
If we're going to try out the clientside call, I'd say we should try to defer it until a certain user action, like the opening of the menu. This way we ensure that the data is only called when the user is in fact is a mobile-esque context. If we move it to the client, but then call it everytime anyways, I think it'd be better to just keep it in the server, if that makes sense.

Reply:
🗒️ Will be deferring the work for optimizing the API call to a follow-up PR.